eidas_node/tests/proxy_service/test_views.py

```diff
@@ -0,0 +1,385 @@
+            saml_request_xml = f.read()
+
+        response = self.client.post(self.url, {'SAMLResponse': b64encode(saml_request_xml).decode('ascii'),
+                                               'RelayState': 'relay123'})
+
+        # Context
+        self.assertIn('token', response.context)
+        self.assertEqual(response.context['token_parameter'], 'test_token')
+        self.assertEqual(response.context['eidas_url'], 'https://test.example.net/SpecificProxyServiceResponse')
+        self.assertEqual(response.context['error'], None)
+
+        # Token
+        encoded_token = response.context['token']
+        token = LightToken.decode(encoded_token, 'sha256', 'response-token-secret')
+        self.assertEqual(token.id, 'uuid4')
+        self.assertEqual(token.issuer, 'response-token-issuer')
+        self.assertEqual(token.created, datetime(2017, 12, 11, 14, 12, 5))
+
+        # Storing light response
+        light_response_data = LIGHT_RESPONSE_DICT.copy()
+        light_response_data.update({
+            'status': Status(**light_response_data['status']),
+            'id': 'uuid4',
+            'in_response_to_id': 'test-light-request-id',
+            'issuer': 'https://test.example.net/node-proxy-service-response',
+        })
+        light_response = LightResponse(**light_response_data)
+        self.assertEqual(self.client_class_mock.mock_calls, [call(timeout=66)])
+        self.assertEqual(self.client_mock.mock_calls,
+                         [call.connect('test.example.net', 1234),
+                          call.get_cache('test-proxy-service-request-cache'),
+                          call.get_cache().get('test-saml-request-id'),
+                          call.get_cache('test-proxy-service-response-cache'),
+                          call.get_cache().put('uuid4', dump_xml(light_response.export_xml()).decode('utf-8'))])
+
+        # Rendering
+        self.assertContains(response, 'Redirect to Service Provider is in progress')
+        self.assertContains(response, '<form action="https://test.example.net/SpecificProxyServiceResponse"')
+        self.assertContains(response, '<input type="hidden" name="test_token" value="{}"'.format(encoded_token))
+        self.assertNotIn(b'An error occurred', response.content)
+
+    def test_post_failure(self):
+        response = self.client.post(self.url)
+        self.assertNotIn(b'https://test.example.net/SpecificProxyServiceResponse', response.content)
+        self.assertContains(response, 'An error occurred', status_code=400)
+        self.assertEqual(response.context['error'].message, 'Bad proxy service request.')
+        self.assertNotIn('eidas_url', response.context)
+        self.assertNotIn('token', response.context)
+        self.assertNotIn('token_parameter', response.context)
+
+
+@override_settings(ROOT_URLCONF='eidas_node.proxy_service.cznia.urls')
+class TestCzNiaResponseView(IgniteMockMixin, SimpleTestCase):
```

Comment:
I would rather have this in a separate file.

Reply:
Moved to a separate file.